lib/spack/llnl/path.py

```diff
@@ -34,7 +35,7 @@ def format_os_path(path: str, mode: int = Path.unix) -> str:
         path = path.replace("/", "\\")
     else:
         path = path.replace("\\", "/")
-    return path
+    return str(path)
```

Comment:
It seems changes to this file can be undone[^1]

[^1]: The function accepts a string, returns a string, no need to add `str()`

Reply:
Yup! No need for the pathlib->string conversion, good catch.
Still need to investigate why pathlib was failing here....'


On another note: is that a footnote with cyclic links? Awesome